### DISCUSSION TITLE
Publishing a release for an old tag

### DISCUSSION THREAD

**Discussion by alvmarrod** [goreleaser/goreleaser#5113] [Q&A] [answered]
I have setup `goreleaser` for [my project](https://github.com/alvmarrod/pi-monitor-api), and both in github and locally the release completes successfully. However, the latest tag avaiable (and only one so far) is not on the latest commit. 

Because of this `tag commit != current commit` condition, I believe my only option is to run the releaser using the flag `--snapshot`, but this (if I understand it correctly) disables the publishing of the results.

```bash
goreleaser release --snapshot --clean
  • skipping announce, publish and validate...
  • cleaning distribution directory
...
```

I have also tried to manually publish this tag, by executing the release manually and locally on the tag commit. However this also fails, as `goreleaser` complains about "git is on dirty state":

```
  • could not find a configuration file, using defaults...
  • loading environment variables
    • using token from  $GITHUB_TOKEN
  • getting and validating git state
    • couldn't find any tags before "1.0.0"
    • git state                                      commit=ac668f88b8003440a0c0005473c9672829b86bc5 branch=HEAD current_tag=1.0.0 previous_tag=<unknown> dirty=true
  ⨯ release failed after 0s                  error=git is in a dirty state
```

Any idea on how could I accomplish the publication of this release?

**Comment by caarlos0** [comment, marked as answer]
you can use `--skip=validate`, which will release anyway

although the better approach is to checkout said tag, and run goreleaser from it.

**Reply by alvmarrod** [reply]
Thanks Carlos, I will try with this option now and get back to you.

About `checkout` and run there, I am afraid I can't do it. As shared in the question, it would fail complaining about `error=git is in a dirty state`. The reason is `goreleaser` has just been setup, so in the tag commit the file `.goreleaser.yaml` didn't exist.

I need this file to specify my `main` location, so if I provide that file the error above would appear again as I can't execute `gorelease release` on the commit without further changes. Sample here for the sake of completeness (again, execution with `.goreleaser.yaml` is the second output block in the question):

```
 goreleaser release
  • could not find a configuration file, using defaults...
  • loading environment variables
    • using token from  $GITHUB_TOKEN
  • getting and validating git state
    • couldn't find any tags before "1.0.0"
    • git state                                      commit=ac668f88b8003440a0c0005473c9672829b86bc5 branch=HEAD current_tag=1.0.0 previous_tag=<unknown> dirty=false
  • parsing tag
  • setting defaults
  • ensuring distribution directory
  • setting up metadata
  • writing release metadata
  • loading go mod information
    • pipe skipped                                   reason=not a go module
  • build prerequisites
  • building binaries
    • building                                       binary=dist/pi-monitor-api_windows_386/pi-monitor-api.exe
    • building                                       binary=dist/pi-monitor-api_linux_386/pi-monitor-api
    • building                                       binary=dist/pi-monitor-api_windows_arm64/pi-monitor-api.exe
    • building                                       binary=dist/pi-monitor-api_windows_amd64_v1/pi-monitor-api.exe
    • building                                       binary=dist/pi-monitor-api_linux_amd64_v1/pi-monitor-api
    • building                                       binary=dist/pi-monitor-api_darwin_arm64/pi-monitor-api
    • building                                       binary=dist/pi-monitor-api_linux_arm64/pi-monitor-api
    • building                                       binary=dist/pi-monitor-api_darwin_amd64_v1/pi-monitor-api
  ⨯ release failed after 0s                  error=build for pi-monitor-api does not contain a main function
Learn more at https://goreleaser.com/errors/no-main
```

**Reply by caarlos0** [reply]
checkout the commit, copy the .goreleaser.yml config file, run it with `--skip=validate` (or move the file outside the git repo)
